extension/chrome/settings/modules/add_key.ts

```diff
@@ -14,63 +14,83 @@ import { Assert } from '../../../js/common/assert.js';
 import { KeyImportUi, UserAlert, KeyCanBeFixed } from '../../../js/common/ui/key_import_ui.js';
 import { initPassphraseToggle } from '../../../js/common/ui/passphrase_ui.js';
 import { Xss } from '../../../js/common/platform/xss.js';
+import { View } from '../../../js/common/view.js';
 
-Catch.try(async () => {
+View.run(class AddKeyView extends View {
 
-  const uncheckedUrlParams = Url.parse(['acctEmail', 'parentTabId']);
-  const acctEmail = Assert.urlParamRequire.string(uncheckedUrlParams, 'acctEmail');
-  const parentTabId = Assert.urlParamRequire.string(uncheckedUrlParams, 'parentTabId');
+  private readonly acctEmail: string;
+  private readonly parentTabId: string;
+  private readonly keyImportUi = new KeyImportUi({ rejectKnown: true });
 
-  await initPassphraseToggle(['input_passphrase']);
-  const keyImportUi = new KeyImportUi({ rejectKnown: true });
-  keyImportUi.initPrvImportSrcForm(acctEmail, parentTabId);
+  constructor() {
+    super();
+    const uncheckedUrlParams = Url.parse(['acctEmail', 'parentTabId']);
+    this.acctEmail = Assert.urlParamRequire.string(uncheckedUrlParams, 'acctEmail');
+    this.parentTabId = Assert.urlParamRequire.string(uncheckedUrlParams, 'parentTabId');
+  }
 
-  Xss.sanitizeRender('#spinner_container', Ui.spinner('green') + ' loading..');
+  async render() {
+    await initPassphraseToggle(['input_passphrase']);
+    this.keyImportUi.initPrvImportSrcForm(this.acctEmail, this.parentTabId);
+    Xss.sanitizeRender('#spinner_container', Ui.spinner('green') + ' loading..');
+    await this.loadAndRenderKeyBackupsOrRenderError();
+    $('.source_selector').css('display', 'block');
+    $('#spinner_container').text('');
+  }
 
-  const keyinfos = await Store.keysGet(acctEmail);
-  const privateKeysLongIds = keyinfos.map(ki => ki.longid);
-  let keyBackups;
+  setHandlers() {
+    $('.action_add_private_key').click(Ui.event.prevent('double', this.addPrivateKeyHandler));
```

Comment:
1) Here please use `this.setHandlerPrenvent` instead.
2) I'm almost certain that this must have broken the functionality of `addPrivateKeyHandler` because `this` would be the HTMLElement, not this class. That means our tests are not sufficient to notice the brokenness. Please manually test it to see if it's broken or not, and assuming it is, please design a test that specifically fails when this functionality is utilized, and which should get fixed when you switch to `this.setHandlerPrevent`

Reply:
Oh, sorry, I didn't notice that there is this type of handlers.
Okay, will do

Reply:
```
I'm almost certain that this must have broken the functionality of addPrivateKeyHandler because this would be the HTMLElement, not this class. That means our tests are not sufficient to notice the brokenness. Please manually test it to see if it's broken or not, and assuming it is, please design a test that specifically fails when this functionality is utilized, and which should get fixed when you switch to this.setHandlerPrevent
```
I think I checked it before but I will double-check it

Reply:
Yes, It worked because I have declared `addPrivateKeyHandler` function like a lamda expression. But I will change the code a little so the `addPrivateKeyHandler` function will be like a method.

Reply:
Ok, that's an interesting solution. I don't like that we would have to think about which methods are lambdas and which are just regular methods (I like consistency). But I do like that you wrote functioning code :)

I will also go ahead and remove the (el) => this... and just directly reference this.xxx in the callback, to make it same as the other files.

Thanks!